frontend-react/src/hooks/api/deliveries/UseDeliveriesHistory/UseDeliveriesHistory.ts

```diff
@@ -0,0 +1,144 @@
+import { useSuspenseQuery } from "@tanstack/react-query";
+import { useCallback, useMemo, useState } from "react";
+import { validate as uuidValidate } from "uuid";
+import {
+    deliveriesEndpoints,
+    RSDeliveryHistoryResponse,
+} from "../../../../config/endpoints/deliveries";
+import useSessionContext from "../../../../contexts/Session/useSessionContext";
+import useFilterManager, {
+    FilterManagerDefaults,
+} from "../../../filters/UseFilterManager/UseFilterManager";
+import useAdminSafeOrganizationName from "../../../UseAdminSafeOrganizationName/UseAdminSafeOrganizationName";
+
+const { getDeliveriesHistory } = deliveriesEndpoints;
+
+export enum DeliveriesDataAttr {
+    REPORT_ID = "reportId",
+    BATCH_READY = "batchReadyAt",
+    EXPIRES = "expires",
+    ITEM_COUNT = "reportItemCount",
+    FILE_NAME = "fileName",
+    RECEIVER = "receiver",
+}
+
+export type SearchParams =
+    | {
+          reportId: string;
+      }
+    | {
+          fileName: string;
+      };
+
+export type SearchFetcher<T> = (
+    additionalParams?: SearchParams,
+) => Promise<T[]>;
+
+const filterManagerDefaults: FilterManagerDefaults = {
+    sortDefaults: {
+        column: DeliveriesDataAttr.BATCH_READY,
+        locally: true,
+    },
+    pageDefaults: {
+        size: 10,
+    },
+};
+
+const useDeliveriesHistory = (initialService?: string) => {
```

Comment:
Any particular reason why we don't take the service name and searchTerm absolutely from the parameters passed instead of having to resort to state? The caller updating either of those would always be reflected.

Reply:
you mean moving:

```js
const [service, setService] = useState(initialService ?? "");
const [searchTerm, setSearchTerm] = useState("");
```

from `UseDeliveryHistory.ts` and just have them passed in? We'd need to maintain them in state somewhere, ie within `DailyData.tsx` and since `UseDeliveryHistory.ts` already is maintaining so much state for the table (page number, sort directions, etc) it would make sense to me to keep it all contained there.

Reply:
It should be the caller's responsibility to hold it's own state of what to pass to a data hook for fetching as it can infinitely vary. The data hook should be as dumb as possible, and it really doesn't need to go an extra step of providing state getters and setters for data it itself doesn't do anything with other than passing it on. We don't have a clear pattern to follow for these things so it doesn't require a rework, but wanted to at least touch on it for later investigation.